pkg/volume/azure_file/azure_file.go

```diff
@@ -225,9 +225,11 @@ func (b *azureFileMounter) SetUpAt(dir string, fsGroup *int64) error {
 		options := []string{fmt.Sprintf("username=%s,password=%s", accountName, accountKey)}
 		if b.readOnly {
 			options = append(options, "ro")
+			// we don't respect fsGroup value if b.readOnly = true
+			fsGroup = nil
```

Comment:
I think we can make use of fsGroup here. Some implementations do mount with ro first, so that gid could not be changed later. 
Here it should be fine as it is passed via mount option,

Reply:
according to https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/aws_ebs/aws_ebs.go#L379-L381
no `fsGroup` value will be set if readOnly = true
```
	if !b.readOnly {
		volume.SetVolumeOwnership(b, fsGroup)
	}
```
Anyway I reserved fsGroup value and use `groupID` var instead, not change the passing parameter value.